ir_actions_report_multi_company/models/ir_actions_actions.py

```diff
@@ -1,28 +1,95 @@
 # Copyright 2022 XCG Consulting
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
-from odoo import api, models, tools
+from odoo import models, tools
+from odoo.tools import frozendict
 
 
-class IrActions(models.Model):
+class IrActionsActions(models.Model):
     _inherit = "ir.actions.actions"
 
     # Add companies to the cache of this method that is used in
     # models.BaseModels.fields_view_get to get the toolbar actions that are now
     # company dependant.
-    @api.model
     @tools.ormcache(
         "frozenset(self.env.user.groups_id.ids)",
         "frozenset(self.env.companies.ids)",
         "model_name",
+        "self.env.lang",
     )
-    def get_bindings(self, model_name):
+    def _get_bindings(self, model_name):
         """Retrieve the list of actions bound to the given model.
 
         :return: a dict mapping binding types to a list of dict describing
                  actions, where the latter is given by calling the method
                  ``read`` on the action record.
         """
+        # get company related reports
+        reports = self._get_report_bindings(model_name)
+
         # Using __wrapped__ is necessary to bypass the cache put by Odoo on the
         # super method.
-        result = super().get_bindings.__wrapped__(self, model_name)
+        result = super()._get_bindings.__wrapped__(self, model_name)
+
+        # update bindings result
+        if reports:
+            result = dict(result)
+            result["report"] = reports
+            result = frozendict(result)
+
+        return result
+
+    def _get_report_bindings(self, model_name):
+        """
+        Get report bindings for current company
+        """
+        cr = self.env.cr
+        self.env.flush_all()
+
+        query = """
+            SELECT report.id, report.type
+              FROM ir_act_report_xml report
+              JOIN ir_model m ON report.binding_model_id = m.id
+             WHERE m.model = %s
+               AND (report.company_id = ANY(%s) OR report.company_id IS NULL)
+               AND report.binding_type = 'report'
+        """
+        cr.execute(query, (model_name, self.env.companies.ids))
+
+        report_ids = [report_id for report_id, report_type in cr.fetchall()]
```

Comment:
`report_type` variable is not used. Please rename `report_type` on `report_type_`.

Reply:
@geomer198 renamed